### DISCUSSION TITLE
ESP32 with Ethernet port, any recommendations?

### DISCUSSION THREAD

**Discussion by OttoMan28** [micropython/micropython#9690] [Hardware & Peripherals]
I can see that over time there has been a bit of debate back and forth about various boards with Ethernet ports, numerous problems have been reported ect.
So, I wonder what’s the status for a working ESP-32 Ethernet board, in Q3-2022 

What can you guys recommend??

Does anyone have good experiences with **Olimex's** ESP32-PoE / ESP32-PoE-ISO ???
In terms of specification, it matches exactly what I'm looking for

- Ethernet
- PoE (but not a requirement)
- WIFI
- Battery charger, LiPo

Thanks in advance.
OM

**Comment by mattytrentini** [comment]
The [wESP32](https://wesp32.com/) is an excellent PoE option though it doesn't have onboard battery charging.

(The original [wESP32 Crowd Supply page](https://www.crowdsupply.com/silicognition/wesp32) has a comparison with the Olimex board. The Olimex is a popular option too but I particularly like that the wESP32 has _excellent_ power isolation.)

**Reply by robert-hh** [reply]
> Does anyone have good experiences with Olimex's ESP32-PoE / ESP32-PoE-ISO

I just made two PRs for these boards, #10329 and #10330. Being PRs, you have to build the firmware yourself until these will be eventually merged. They are based on a ESP32  WROOM module: 4 MB flash, no SPIRAM.
A sample firmware build is here: https://github.com/robert-hh/Shared-Stuff/blob/master/firmware_OLIMEX_ESP32_POE.bin
